docs/DevGuide/Databox.md

```diff
@@ -0,0 +1,367 @@
+subsequent quantities. Let us take a second look at `force_compute`. `mass` and
+`acceleration` are obtained via a computation. This means that `mass` and
+`acceleration` are recomputed every time `force_compute` is called. If `Mass`
+and `Acceleration` were tags somewhow, that is, if we could compute them once,
+place them in the databox, and get them back out through the use of tags, we
+could get around this problem. We are now ready to consider SpECTRE's DataBox,
+which provides the solution to this problem in the form of `ComputeTags`.
+
+# SpECTRE's DataBox {#a_proper_databox}
+
+A brief description of SpECTRE's DataBox: a TaggedTuple with compute-on-demand.
+For a detailed description of SpECTRE's DataBox, see the
+\ref DataBoxGroup "DataBox documentation".
+
+## SimpleTags {#documentation_for_simple_tags}
+Just as we needed to modify our tags to make them compatible with
+`TaggedTuple`, we need to again modify them for use with DataBox. Our ordinary
+tags become SpECTRE's SimpleTags:
+
+\snippet Test_DataBoxDocumentation.cpp proper_databox_tags
+
+As seen above, SimpleTags have a `type` and a `name` in their struct.
+When creating tags for use with a DataBox, we must make sure to derive off
+of one of the existing DataBox tag types such as `db::SimpleTag`. We now
+create our first DataBox using these `SimpleTags`:
+
+\snippet Test_DataBoxDocumentation.cpp refined_databox
+
+We can get our quantities out of the DataBox by using `db::get`:
+
+\snippet Test_DataBoxDocumentation.cpp refined_databox_get
+
+So far, the usage of DataBox has been similar to the usage of TaggedTuple. To
+address the desire to combine the functionality of tags with the modularity of
+functions, DataBox provides ComputeTags.
+
+## ComputeTags {#documentation_for_compute_tags}
+ComputeTags are used to tag functions that are used in conjunction with a
+DataBox to produce a new quantity. ComputeTags look like:
+
+\snippet Test_DataBoxDocumentation.cpp compute_tags
+
+ComputeTags derive off of `db::ComputeTag`, and it is convenient to have them
+additionally derive off of an existing SimpleTag (in this case `Mass`) so that
+the quantity `MassCompute` can be obtained through the SimpleTag `Mass`.
+We use the naming convention `TagNameCompute` so that `TagNameCompute` and
+`TagName` appear next to each other in documentation that lists tags in
+alphabetical order.
+
+We have also added the type alias `argument_tags`, which is necessary in order
+to refer to the correct tagged quantities in the DataBox.
+
+\note
+The `tmpl::list` used in the type alias is a contiguous container only
+holding types. That is, there is no variable runtime data associated with it
+like there is for `std::tuple`, which is a container associating types with
+values. They are useful in situations when one is working with multiple tags at
+once.
+
+This compile time resolution of *metavariables* is a pattern that is used again
```

Comment:
Using nested type aliases to pass around information at compile time is common pattern in SpECTRE